Test/functional/Test_State_Machine.cpp

```diff
@@ -332,13 +342,144 @@ TEST_F(StateTest, Pressure_Vessel_Depressure_Test) {
  * Test the nominal transition from idle to pumpdown
  * Start state: idle        Expected state: pumpdown
  */
-TEST_F(StateTest, Idle_To_Pumpdown_Test) {
+TEST_F(StateTest, Idle_To_Pumpdown_Test)
+{
     FREEZE_SM;
     genericInit();
     fprintf(stderr, "[LOG] Going into Pumpdown\n");
     GO_TO_STATE(PUMPDOWN_NAME);
     UNFREEZE_SM;
     WAIT(.5);
+    ASSERT_EQ(getFlagsShouldBrake(), false);
+
+    debuggingHelper();
 
     assertStateIs(PUMPDOWN_NAME);
-}
\ No newline at end of file
+}
+/**
+ * Test the nominal transition from idle to pumpdown
+ * Start state: pumpdown        Expected state: propulsion
+ */
+TEST_F(StateTest, Pumpdown_to_Propulsion_Test)
+{
+    FREEZE_SM;
+    genericInit();
+    fprintf(stderr, "[LOG] Going into Propulsion\n");
+    GO_TO_STATE(PUMPDOWN_NAME);
+    GO_TO_STATE(PROPULSION_NAME);
```

Comment:
You want to make sure you unfreeze the state machine first - then change the state. This is why the tests are failing. Do this for all of the tests.